storage/src/main/java/tech/pegasys/teku/storage/server/StorageConfiguration.java

```diff
@@ -289,7 +289,7 @@ public Builder statePruningInterval(final Duration statePruningInterval) {
     }
 
     public Builder statePruningLimit(final int statePruningLimit) {
-      if (statePruningLimit < 0) {
```

Comment:
NIT: I'd add a static variable named `MAX_STATE_PRUNING_LIMIT` with a comment about the value 100 being an arbitrary choice. We recently had users questioning some of our choices for default values and limits, and I think making it explicit that it was arbitrary can help us remember in the future.

Reply:
I'm still testing this PR to get a better sense of what is a more sensible limit.

Reply:
done and tests have confirmed that 100 limit is safe in the current instance we use for mainnet archive